### PR TITLE
feat: update models list ui/ux

### PR DESCRIPTION
**Pull request by orangecoloured**
[Figma](https://www.figma.com/file/QZtPfI8NDuhEv5lPYX4z2a/Instill-Model?type=design&node-id=11%3A5322&mode=design&t=14khhkK5vHqHsh00-1)

[Eraser](https://app.eraser.io/workspace/3ZrAWVNuPMCg6QSpuD9s?origin=share)

It's a bit dirty. But most of the changes related to the models list.

### CODE REVIEW

**Comment by orangecoloured** [comment]
@EiffelFly we can't jump to pages and their component doesn't seem to have the "1 of 2" indicator. For me it feels like replacing a couple of elements and the rest of the component will stay the same. Basically I will just replace the `Button`s. Easy. Just wondering whether it makes sense with current functionality.

**Comment by EiffelFly** [comment]
> we can't jump to pages and their component doesn't seem to have the "1 of 2" indicator.

@orangecoloured Fair point, in this case we don't need to put it into design-system due to it's not universal enough. If we still want to put it into design-system then we need to think in long-term. 

- The component should stay in the composition component manner
- The component need to be future proof

That is the reason I mentioned the shacdn/ui's pagination 🦈 

I think it's on your call. We can definitely let it stay in toolkit as a small component, or we implement shacdn/ui's component in design-system and adapt it in the model list page

**Comment by orangecoloured** [comment]
@EiffelFly refactored the Pagination component.

**Comment by EiffelFly** [comment]
@orangecoloured Hi Alex, please make sure the PR is at least buildable 🦈

File: packages/toolkit/src/lib/react-query-service/model/useModels.ts
Comment (EiffelFly) [inline packages/toolkit/src/lib/react-query-service/model/useModels.ts:6]: @orangecoloured Due to this behavior is different from normal useModels, we should maintain two set of queries, one is normal useModels. And another one will be `useInfiniteModels`

File: packages/toolkit/src/lib/vdp-sdk/model/types.ts
Comment (EiffelFly) [inline packages/toolkit/src/lib/vdp-sdk/model/types.ts:43]: @orangecoloured Do you think it's better to use Record<string, string> here?

File: packages/toolkit/src/view/model/CreateModelForm1.tsx
Comment (EiffelFly) [inline packages/toolkit/src/view/model/CreateModelForm1.tsx:22]: @orangecoloured Could you help me extract these enum into the constant folder, in this way we can make it more maintainable

File: packages/toolkit/src/view/model/CreateModelForm1.tsx
Comment (EiffelFly) [inline packages/toolkit/src/view/model/CreateModelForm1.tsx:52]: @orangecoloured Is this finished?

**Review by EiffelFly — COMMENTED** [review]
@orangecoloured It seems like the feature is not ready, so I comment some code that I think it's worth discussing

File: packages/toolkit/src/lib/react-query-service/model/useModels.ts
Comment (orangecoloured) [reply packages/toolkit/src/lib/react-query-service/model/useModels.ts:6]: Should we only have `useInfiniteModels` then? It's a paginated endpoint and feels like it doesn't make sense to have it without pagination.

File: packages/toolkit/src/lib/vdp-sdk/model/types.ts
Comment (orangecoloured) [reply packages/toolkit/src/lib/vdp-sdk/model/types.ts:43]: Technically, I see no difference. But `Record` is shorter and feels more "typed". I fixed it.

File: packages/toolkit/src/view/model/CreateModelForm1.tsx
Comment (orangecoloured) [reply packages/toolkit/src/view/model/CreateModelForm1.tsx]: This is finished in the other PR. This is the dirty part I mentioned. I touched some "create" stuff in this PR.

File: packages/toolkit/src/view/model/CreateModelForm1.tsx
Comment (orangecoloured) [reply packages/toolkit/src/view/model/CreateModelForm1.tsx]: Let's resolve this in the next PR. Sorry I spilled some create form work into this PR.

File: packages/toolkit/src/lib/react-query-service/model/useModels.ts
Comment (EiffelFly) [reply packages/toolkit/src/lib/react-query-service/model/useModels.ts:6]: Yes we can do that before we have other use cases.

Comment (orangecoloured) [reply packages/toolkit/src/lib/react-query-service/model/useModels.ts]: Done.

File: packages/toolkit/src/lib/react-query-service/model/useModels.ts
Comment (EiffelFly) [inline packages/toolkit/src/lib/react-query-service/model/useModels.ts:1]: @orangecoloured The filename should be renamed to `useInfiniteModels`

File: packages/toolkit/src/view/model/ModelHubListPageMainView.tsx
Comment (EiffelFly) [inline packages/toolkit/src/view/model/ModelHubListPageMainView.tsx:126]: @orangecoloured react-query will make sure once the models is success, the data is there, so you don't need to use `Optional chaining` here

File: packages/toolkit/src/view/model/ModelsList.tsx
Comment (EiffelFly) [inline packages/toolkit/src/view/model/ModelsList.tsx:24]: @orangecoloured We have accessToken and enbaedQuery in our global store. 

You can access them with useInstillStore, please look for pipeline related component for how to use it

**Review by EiffelFly — COMMENTED** [review]
@orangecoloured I think generally speaking the code quality is nice, I left some minor stuff in the comment, please check them

File: packages/toolkit/src/view/model/CreateModelForm.tsx
Comment (orangecoloured) [reply packages/toolkit/src/view/model/CreateModelForm.tsx:508]: This is actually completely removed in the other PR. But ok about rel. I just thought that if it's our page then it shouldn't really matter. Will be adding it in the future.

File: packages/toolkit/src/view/model/CreateModelForm1.tsx
Comment (orangecoloured) [reply packages/toolkit/src/view/model/CreateModelForm1.tsx:68]: Could we just dismiss this file? It's removed in the other PR.
Again, sorry I left it like this.

File: packages/toolkit/src/view/model/ModelHubListPageMainView.tsx
Comment (orangecoloured) [reply packages/toolkit/src/view/model/ModelHubListPageMainView.tsx:126]: Do you mean to have it like this?
```
models={
  models.isSuccess
    ? models.data.pages[pageNumber]?.models
    : []
}
```

Comment (EiffelFly) [reply packages/toolkit/src/view/model/ModelHubListPageMainView.tsx:126]: yes exactly

File: packages/toolkit/src/view/model/ModelHubCreatePageMainView.tsx
Comment (EiffelFly) [inline packages/toolkit/src/view/model/ModelHubCreatePageMainView.tsx:26]: @orangecoloured I think this a dev changes, I think we need to change it back

File: packages/toolkit/src/view/model/ModelsList.tsx
Comment (EiffelFly) [inline packages/toolkit/src/view/model/ModelsList.tsx:31]: @orangecoloured If we don't need the state here, we need to remove it from the selector

File: packages/toolkit/src/components/card-model/CardModel.tsx
Comment (EiffelFly) [inline packages/toolkit/src/components/card-model/CardModel.tsx:36]: @orangecoloured I think this callback return is a bit confusing, I noticed that you try to use `query.refetch`

We can make it more general, just like (options) => void

and you use it as `() => query.refetch`

But if you want this prop to be only used for refetch then we should rename it to be more specific. I am ok with either way 🦤

File: packages/toolkit/src/view/model/CreateModelForm1.tsx
Comment (EiffelFly) [reply packages/toolkit/src/view/model/CreateModelForm1.tsx:68]: But since this PR is merged into the main, and since you already have this one another PR, I think we can safely delete it?

**Review by EiffelFly — COMMENTED** [review]
@orangecoloured I think the code wise it's good now, I left some comment and I will begin to review the functionality

File: packages/toolkit/src/view/model/CreateModelForm1.tsx
Comment (orangecoloured) [reply packages/toolkit/src/view/model/CreateModelForm1.tsx:68]: Alright. Makes sense. Done.

File: packages/toolkit/src/view/model/ModelHubCreatePageMainView.tsx
Comment (orangecoloured) [reply packages/toolkit/src/view/model/ModelHubCreatePageMainView.tsx:26]: I removed it.

File: packages/toolkit/src/components/card-model/CardModel.tsx
Comment (orangecoloured) [reply packages/toolkit/src/components/card-model/CardModel.tsx:36]: I simplified it to `() => void`. If we need to pass any options we can always update the type.

**Review by EiffelFly — COMMENTED** [review]
@orangecoloured I left some comments please check them. Once Heiru merged his PR into main I will begin to review the functionality and style

File: packages/toolkit/src/components/card-model/CardModel.tsx
Comment (EiffelFly) [inline packages/toolkit/src/components/card-model/CardModel.tsx:32]: @orangecoloured We don't directly interpret entity like this since there will be entity for user and organization. That is the reason we rely on useAppEntity to handle this stuff 🦤

**Review by EiffelFly — COMMENTED** [review]
@orangecoloured Final comment!! We are very close 🦤

**Comment by EiffelFly** [comment]
@orangecoloured There is one issue that need to be solved in this PR, it's about the visibility of model on the /entity/models page.

## Let's make it act like this way 

- When the viewer is exact the owner of the current entity, the /entity/models page should display all the model owned by the entity
- When the viewer is not the owner of the current entity, the /entity/models page should display the public models owned by the entity

## For example

- Current login user's entity is summerbud, I belong to the org instill-ai
- When I visit /summerbud/models -> Display all the models I own
- When I visit /instill-ai/models -> Display all the models instill-ai own
- When I visit /alex/models -> Display all the models Alex own
- When I visit /alex-corp/models -> Display all the public models Alex own

## How to implement this

We have a similar endpoint for pipelines https://github.com/instill-ai/console/blob/121c5704f036db205808e6bd1a6e1de8b93f1530/packages/toolkit/src/lib/vdp-sdk/pipeline/queries.ts#L114

So you will need additional endpoint which query the models under specific entity like `getUserModelsQuery`

At the same time we need to make it act as infinite-query to enable current UX on page

**Comment by EiffelFly** [comment]
@orangecoloured Some UI issues I discovered

## We need to make the search and visibility selection fixed width
according to the design handover they are both `w-[300px]`

https://github.com/instill-ai/console/assets/57251712/b415d44b-a380-47c6-9010-f3d3a4083206

## The create model 
- The create model page's content should has a fixed width 
- The margin of this page is not correct. 
- The `Create a model` font style is not correct
- When I expand the width of the screen the content should has fixed width and centered

https://github.com/instill-ai/console/assets/57251712/0613dd40-e434-40b4-a35a-f6df5ee7be9a

## The version tab placeholder
There has some flicker happening when the version is loading


https://github.com/instill-ai/console/assets/57251712/b5d44ba6-145e-4e05-a43d-a320f523cc63

**Comment by EiffelFly** [comment]
@orangecoloured I forgot to mention the URL structure might need to align with the tab structure

For example, when the user switch to version tab the URL should be /summerbud/models/mid/versions

So user can quicker navigate to it from different space

**Review by EiffelFly — COMMENTED** [review]
Hi Alex, I only have this one question left

File: packages/toolkit/src/components/ImageWithFallback.tsx
Comment (orangecoloured) [reply packages/toolkit/src/components/ImageWithFallback.tsx:33]: The use case:

We have an image which is broken and the component switches into error mode. We update the image, for example using a modal or some other way wich does not require to change routes or dismount the component. Being in the error mode nothing will change and to see the updated image we need to reset the component, either refresh the page or switch views etc. With this logic as soon as the image component receives a new image src, it tries to render the image again. And if succeedes - we see an updated image, if not - it falls back into the error mode.

**Review by EiffelFly — DISMISSED** [review]
LGTM

**Review by EiffelFly — APPROVED** [review]
LGTM

**Pull request merged**
